### PR TITLE
[ADD] account_invoice_tax_required

### PR DESCRIPTION
**Pull request by vrenaville**
If you need that All lines of your invoice have a tax when you validate it

### CODE REVIEW

File: account_invoice_tax_required/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': "Tax required in invoice",
+    'version': "1.0",
+    'author': "Vincent Renaville",
```
Comment (adrienpeiffer) [inline account_invoice_tax_required/__openerp__.py]: Can you add OCA as author ?

**Comment by adrienpeiffer** [comment]
You should remove the xml file

File: account_invoice_tax_required/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        # check whether all corresponding account move lines are reconciled
+        for invoice in self:
+            for invoice_line in invoice.invoice_line:
+                #
+                if not invoice_line.invoice_line_tax_id:
+                    error_string = error_template % (invoice_line.name)
+                    errors.append(error_string)
+        if errors:
+            errors_full_string = ','.join(x for x in errors)
+            raise except_orm(_('No Taxes Defined!'), errors_full_string)
```
Comment (sbidoul) [inline account_invoice_tax_required/models/account_invoice.py]: except_orm is deprecated, raise exceptions.Warning instead

File: account_invoice_tax_required/README.rst [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+to force user to set tax on invoice line.
+
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/account-financial-tools/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
```
Comment (adrienpeiffer) [inline account_invoice_tax_required/README.rst:14]: Issue creating link is missing

File: account_invoice_tax_required/models/account_invoice.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+from openerp.exceptions import except_orm
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.multi
+    def test_invoice_line_tax(self):
+        errors = []
+        error_template = _("Invoice have a line with product %s with no taxes")
+        # check whether all corresponding account move lines are reconciled
```
Comment (adrienpeiffer) [inline account_invoice_tax_required/models/account_invoice.py]: You should remove this comment

**Comment by fclementic2c** [comment]
functional test OK
:+1:

**Comment by vrenaville** [comment]
@adrienpeiffer @sbidoul Thanks for the review !, I have update my PR according to your comments

File: account_invoice_tax_required/models/account_invoice.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+from openerp import models, api, exceptions, _
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.multi
+    def test_invoice_line_tax(self):
+        errors = []
+        error_template = _("Invoice have a line with product %s with no taxes")
```
Comment (sbidoul) [inline account_invoice_tax_required/models/account_invoice.py]: Invoice should be prural or change 'have' to 'has'.

Or say "Invoice line '%s' has no taxes."

**Comment by vrenaville** [comment]
@sbidoul thanks for typo, I have just correct it.

**Comment by sbidoul** [comment]
@vrenaville there is still this strange account_invoice_view.xml that should be removed.

**Comment by sbidoul** [comment]
:+1:  (functional test and code review)

**Comment by vrenaville** [comment]
@sbidoul sorry for that, I have just remove it.

**Comment by sbidoul** [comment]
:+1:

**Comment by adrienpeiffer** [comment]
There is still `{version}` in the bug tracker. Otherwise :+1:

**Comment by vrenaville** [comment]
@adrienpeiffer I finally correct the version tag
thanks for the review and the time

**Pull request merged**